src/providers/http.ts

```diff
@@ -261,16 +270,36 @@ export async function createTransformRequest(
         functionName,
       );
       if (typeof requiredModule === 'function') {
-        return requiredModule;
+        return (prompt) => {
+          try {
+            return requiredModule(prompt);
```

Comment:
This doesn't work, and neither does the one on line 294.  You need a try/catch in the function itself at call time